sdk/core/core/inc/az_json_string.h

```diff
@@ -15,6 +15,24 @@
 AZ_NODISCARD az_result
 az_span_reader_read_json_string_char(az_span_reader * const self, uint32_t * const out);
 
+/**
+ * Encodes the given character into a JSON escape sequence. The function returns an empty span if
+ * the given character doesn't require to be escaped.
+ */
+AZ_NODISCARD az_span az_json_esc_encode(az_result_byte const c);
+
+enum {
+  AZ_HEX_LOWER_OFFSET = 'a' - 10,
+  AZ_HEX_UPPER_OFFSET = 'A' - 10,
+};
+
+/**
+ * @digit should be in the range [0..15].
+ */
+AZ_NODISCARD AZ_INLINE uint8_t az_digit_to_upper_hex(uint8_t const digit) {
```

Comment:
If this function is public, the name is a bit confusing, I think, `int4 to upper hex` describes it better.
And also if this is public, we probably should make 'to lower hex' version as well.

Reply:
I've renamed it to `az_number_to_upper_hex`.